cli/flox/src/main.rs

```diff
@@ -50,6 +50,10 @@ fn main() -> ExitCode {
     // Avoid SIGPIPE from killing the process
     reset_sigpipe();
 
+    // Eagerly evaluate version and prevent it from propagating to sub-processes.
+    let _ = *FLOX_VERSION_STRING;
+    env::remove_var("FLOX_VERSION");
```

Comment:
Eew, writing around in the environment /s

More seriously, we will lose the flox_version in activations which can be useful to detect activations of old flox versions, although it should then probably be just an entirely different variable.

Reply:
Oh also if you want to make it easier to port to rust 2024 could you put this in an `unsafe { ... }` block

```rust
// SAFETY: Writing to the environment is safe here since we can guarantee
// not to look up the env concurrently,
// because at this point the program is still single threaded.
unsafe { env::remove_var("FLOX_VERSION"); }
```

Reply:
> More seriously, we will lose the flox_version in activations which can be useful to detect activations of old flox versions, although it should then probably be just an entirely different variable.

It's a nice-to-have but I've never actually seen it used, it's nearly always wrong for me because my terminal emulator comes from an outer activation, and whilst we support these three different ways that the version can be generated/passed it feels safer to just prevent it leaking - much like we are doing for environment variables in activations.